File: utio/models/ControlLog.js
/**
 * 控制操作日志模型
 * 对应表：control_logs
 */

class ControlLog {
  constructor(data = {}) {
    this.id = data.id || null
    this.controlType = data.control_type || data.controlType || ''
    this.action = data.action || ''
    this.status = data.status || 'success'
    this.message = data.message || null
    this.createdAt = data.created_at || data.createdAt || null
  }

  /**
   * 转换为数据库字段格式
   */
  toDbFormat() {
    return {
      id: this.id,
      control_type: this.controlType,
      action: this.action,
      status: this.status,
      message: this.message,
    }
  }

  /**
   * 转换为前端格式
   */
  toFrontendFormat() {
    return {
      id: this.id,
      controlType: this.controlType,
      action: this.action,
      status: this.status,
      message: this.message,
      createdAt: this.createdAt,
    }
  }

  /**
   * 验证数据有效性
   */
  validate() {
    const errors = []
    if (!this.controlType || this.controlType.trim() === '') {
      errors.push('控制类型不能为空')
    }
    if (!this.action || this.action.trim() === '') {
      errors.push('操作动作不能为空')
    }
    if (!['success', 'failed'].includes(this.status)) {
      errors.push('操作状态必须是 success 或 failed')
    }
    return errors
  }
}

module.exports = ControlLog

File: utio/springboot/src/main/java/com/greenhouse/entity/ControlLog.java
package com.greenhouse.entity;

import jakarta.persistence.*;
import lombok.Data;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;

/**
 * 控制操作日志实体类
 * 对应表：control_logs
 */
@Entity
@Table(name = "control_logs", indexes = {
    @Index(name = "idx_control_type", columnList = "control_type"),
    @Index(name = "idx_created_at", columnList = "created_at")
})
@Data
@EntityListeners(AuditingEntityListener.class)
public class ControlLog {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    /**
     * 控制类型（light/cleaning等）
     */
    @Column(name = "control_type", nullable = false, length = 50)
    private String controlType;

    /**
     * 操作动作（on/off/start/stop等）
     */
    @Column(nullable = false, length = 50)
    private String action;

    /**
     * 操作状态（success/failed）
     */
    @Column(nullable = false, length = 20)
    private String status;

    /**
     * 操作消息
     */
    @Column(length = 200)
    private String message;

    @CreatedDate
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;
}

